### DISCUSSION TITLE
Continuous Sniffing with Stop Trigger

### DISCUSSION THREAD

**Discussion by ryankshah** [KimiNewt/pyshark#560] [Q&A] [answered]
Hi all!

I am currently aiming to use this library to do a live capture of some traffic related to a program I'm writing. Inherently, the program does some network related actions and the task at hand is akin to the following:

> Start tshark recording, i.e. via:
>>     capture = pyshark.LiveCapture(interface='wlan0', output_file='path/to/file');
>>     capture.sniff_continuously()
> Do some actions which impact network traffic
> Stop recording
> *_Repeat_*

However, from the documentation this seems unclear as to whether it can be accomplished or not!

Any tips/advice on how I could achieve this would be great, thanks!

**Comment by KimiNewt** [comment, marked as answer]
If you use `apply_on_packets` rather than `sniff_continuously` then you can give a callback which can raise `StopCapture` to stop the capture. Then you can call `capture.close()` and reopen it
